src/lava/proc/conv/models.py

```diff
@@ -19,22 +19,33 @@ class AbstractPyConvModel(PyLoihiProcessModel):
     """Abstract template implementation of PyConvModel."""
     s_in = None
     a_out = None
+    a_buf = None
     weight = None
 
     kernel_size: np.ndarray = LavaPyType(np.ndarray, np.int8, precision=8)
     stride: np.ndarray = LavaPyType(np.ndarray, np.int8, precision=8)
     padding: np.ndarray = LavaPyType(np.ndarray, np.int8, precision=8)
     dilation: np.ndarray = LavaPyType(np.ndarray, np.int8, precision=8)
     groups: np.ndarray = LavaPyType(np.ndarray, np.int8, precision=8)
+    use_graded_spike: np.ndarray = LavaPyType(np.ndarray, bool, precision=1)
 
     def run_spk(self) -> None:
-        s_in = self.s_in.recv()
+        if self.use_graded_spike.item():
+            s_in = self.s_in.recv()
+        else:
+            s_in = self.s_in.recv().astype(bool)
+
```

Comment:
I'm still personally in favor of not changing the implementation (or data type) between binary and graded spikes for code simplicity. Rather we'd just encode everything as ints, where binary spikes are 0 or 1. 

But I suppose this involves a check to ensure s_in does not contain any values except 0 and 1 if the user specifies 'use_graded_spike' is False. Maybe this is less efficient than your implementation above?

Reply:
`use_graded_spike` flag is important. It is what decides how the input axon are configured on the Neurocore. I also think changing the data type to bool is good. It avoids unexpected behavior when binary spike is expected and get non-binary value.

I am okay with unifying the computation. For large dense connection one would use sparse connection. This would not matter for small dense connection.

Reply:
Fair enough. We can keep bool for binary and int/float for graded. If things get too hairy or we don't like code duplications, we can consider refactoring.